### PR TITLE
feat:[CDS-70330]: junit PR for tf encryption

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
Junit PR 
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49718)

### CODE REVIEW

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49718)* status: 0 of 1 approvals obtained.

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/terragrunt/TerragruntTaskServiceTest.java [hunk 1]
```diff
@@ -251,4 +255,19 @@ public void testCleanDirectoryAndSecrets() throws IOException {
...
+  public void testSaveTerraformPlanContentToFile() throws IOException {
+    byte[] decryptedTerraformPlan = "Tfplan".getBytes(StandardCharsets.UTF_8);
+
+    when(harnessSMEncryptionDecryptionHandlerNG.getDecryptedContent(any(), any(), any()))
+        .thenReturn(decryptedTerraformPlan);
+
+    taskService.saveTerraformPlanContentToFile(VaultConfig.builder().build(), EncryptedRecordData.builder().build(),
+        "Script", ACCOUNT_ID, TG_TEST_BASE_DIR, true, true);
+
+    verify(harnessSMEncryptionDecryptionHandlerNG, times(1)).getDecryptedContent(any(), any(), any());
```
Comment (vlica-harness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/terragrunt/TerragruntTaskServiceTest.java:271]: if something gets saved to a file, shoudn't we clean after this that file from this path TG_TEST_BASE_DIR?

Comment (sourabh-awashti) [reply 930-delegate-tasks/src/test/java/io/harness/delegate/task/terragrunt/TerragruntTaskServiceTest.java:271]: done

**Comment by sourabh-awashti** [comment]
trigger uts

**Comment by sourabh-awashti** [comment]
trigger uts

**Comment by sourabh-awashti** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 3 of 5 files at r2, 3 of 23 files at r3, 16 of 16 files at r4, 6 of 6 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49718)* status: 31 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by vlica-harness — APPROVED** [review]

**Comment by sourabh-awashti** [comment]
trigger review

**Pull request merged**